openedx/openedx-platform
# Fix versioned URLs for XBlock handlers in v2 runtime [FC-0062]

Fixes https://github.com/openedx/edx-platform/issues/35673

Handlers weren't working in some cases if the handler URL contained a query string, due to assumptions made in the capa XBlock JS (and probably others).

This PR moves the version specifier into the URL path, instead of the query string. Also made some other cleanups.

How to test:
1. Create a multiple choice problem in a library and copy it
2. Paste it into a course
3. Edit the problem and publish the changes
4. Press the "update" button in the course, and see the comparison modal.
5. Try submitting the question in the comparison modal. With the fixes from this PR, it should work.

![Screenshot](https://github.com/user-attachments/assets/7a49f975-89cc-4518-a933-593ed3ff3a54)
